R/runs.R

```diff
@@ -0,0 +1,36 @@
+#' Runs
+#'
+#' @description
+#' - `vec_identify_runs()` returns an identifier for the run of repeated values
+#'   that each element of `x` falls in. The number of runs is also returned as
+#'   an attribute, `n`.
+#'
+#' @details
+#' Unlike [base::rle()], adjacent missing values are considered identical when
+#' constructing runs. For example, `vec_identify_runs(c(NA, NA))` will return
+#' `c(1, 1)`, not `c(1, 2)`.
+#'
+#' @param x A vector.
+#'
+#' @return
+#' - `vec_identify_runs()`: An integer vector with the same size as `x`. A
+#'   scalar integer attribute, `n`, is attached.
+#'
+#' @export
+#' @examples
```

Comment:
I was thinking about adding `vec_partition(x, by)` which would be like `vec_split(x, by)` but would partition `x` using the runs of `by`. Then I'd add examples of that. It would return the same data structure as `vec_split()` (2-col df).

It might be a bit easier to implement vs `vec_partition(x, runs)` since you don't have to validate that the `runs` are correct

Reply:
I was thinking `vec_partition(letters, c(6, 15))` might be generally useful.

Reply:
Possibly room for `vec_partition(x, by)` and `vec_partition_at(x, loc)` (Although I'd imagine that `loc` here would be the starts, not the ends of each run)